src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/JointGermlineCNVSegmentation.java

```diff
@@ -0,0 +1,483 @@
+                        genotypeBuilder.attribute(GermlineCNVSegmentVariantComposer.CN, sampleCopyNumbers.get(sample).getLeft());
+                    }
+                }
+                newGenotypes.add(genotypeBuilder.make());
+            }
+            alleleNumber += samplePloidy;
+        }
+        builder.genotypes(newGenotypes);
+        if (alleleNumber > 0) {
+            if (vc.getAlternateAlleles().size() == 1) {
+                final long AC;
+                if (vc.getAlternateAllele(0).equals(GATKSVVCFConstants.DUP_ALLELE)) {
+                    AC = alleleCountMap.get(GATKSVVCFConstants.DUP_ALLELE);
+                } else {
+                    AC = alleleCountMap.get(GATKSVVCFConstants.DEL_ALLELE);
+                }
+                builder.attribute(VCFConstants.ALLELE_COUNT_KEY, AC)
+                        .attribute(VCFConstants.ALLELE_FREQUENCY_KEY, Double.valueOf(AC) / alleleNumber)
+                        .attribute(VCFConstants.ALLELE_NUMBER_KEY, alleleNumber);
+            } else {
+                final List<Long> alleleCounts = new ArrayList<>();
+                final List<Double> alleleFreqs = new ArrayList<>();
+                //if we merged and del and a dupe from different callsets, then make sure the VC has both alleles
+                //if (alleleCountMap.get(GATKSVVCFConstants.DEL_ALLELE) > 0 && alleleCountMap.get(GATKSVVCFConstants.DUP_ALLELE) > 0) {
+                //    builder.alleles(vc.getReference().getDisplayString(), GATKSVVCFConstants.DUP_ALLELE.getDisplayString(), GATKSVVCFConstants.DEL_ALLELE.getDisplayString());
+                //}
+                for (final Allele a : builder.getAlleles()) {
+                    if (a.isReference()) {
+                        continue;
+                    }
+                    alleleCounts.add(alleleCountMap.containsKey(a) ? alleleCountMap.get(a) : 0L);
+                    alleleFreqs.add(alleleCountMap.containsKey(a) ? Double.valueOf(alleleCountMap.get(a)) / alleleNumber : 0L);
+                }
+                builder.attribute(VCFConstants.ALLELE_COUNT_KEY, alleleCounts)
+                    .attribute(VCFConstants.ALLELE_FREQUENCY_KEY, alleleFreqs)
+                    .attribute(VCFConstants.ALLELE_NUMBER_KEY, alleleNumber);
+            }
+        }
+        return builder.make();
+    }
+
+    /**
+     *
+     * @param sampleName
+     * @param contig
+     * @param g may be null
+     * @return
+     */
+    private int getSamplePloidy(final String sampleName, final String contig, final Genotype g) {
+        if (!allosomalContigList.contains(contig)) {
+            return refAutosomalCopyNumber;
+        }
+        int samplePloidy = 1;
+        if (sampleDB == null || sampleDB.getSample(sampleName) == null) {
+            if (g != null) {
+                samplePloidy = g.getPloidy();
+            } else {
+                oneShotLogger.warn("Samples missing from pedigree assumed to have ploidy 1 on allosomes.");
```

Comment:
I went through a couple different iterations of trying to avoid the ped file, which is now required.  The better solution is just to require all samples to be in the pedigree and check on startup, which I will do.  I don't love that you need a pedigree even if you're just calling over the autosomes, though.  Although maybe we should just go with @samuelklee's philosophy that these inputs are easy if the tool is run in the WDL.